modules/ui_patterns_config/src/Entity/UiPatternsConfig.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+namespace Drupal\ui_patterns_config\Entity;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Serialization\Yaml;
+use Drupal\file\Entity\File;
+use Drupal\ui_patterns_config\UiPatternsConfigInterface;
+
+/**
+ * Defines the UI Patterns configuration entity.
+ *
+ * @ConfigEntityType(
+ *   id = "ui_patterns_config",
+ *   label = @Translation("pattern configuration"),
+ *   handlers = {
+ *     "list_builder" = "Drupal\ui_patterns_config\Controller\UiPatternsConfigListBuilder",
+ *     "form" = {
+ *       "add" = "Drupal\ui_patterns_config\Form\UiPatternsConfigForm",
+ *       "edit" = "Drupal\ui_patterns_config\Form\UiPatternsConfigForm",
+ *       "delete" = "Drupal\ui_patterns_config\Form\UiPatternsConfigDeleteForm",
+ *     }
+ *   },
+ *   config_prefix = "ui_patterns_config",
+ *   admin_permission = "administer UI patterns configuration entities",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *   },
+ *   links = {
+ *     "edit-form" = "/admin/config/system/ui_patterns_config/{ui_patterns_config}",
+ *     "delete-form" = "/admin/config/system/ui_patterns_config/{ui_patterns_config}/delete",
+ *   }
+ * )
+ */
+class UiPatternsConfig extends ConfigEntityBase implements UiPatternsConfigInterface {
+
+  /**
+   * The pattern ID.
+   *
+   * @var string
+   *   The pattern ID.
+   */
+  public $id;
+
+  /**
+   * The pattern label.
+   *
+   * @var string
+   *   The pattern label.
+   */
+  public $label;
+
+  /**
+   * The pattern definition.
+   *
+   * @var string
+   *   The pattern definition.
+   */
+  public $definition;
+
+  /**
+   * The pattern template.
+   *
+   * @var string
+   *   The pattern template.
+   */
+  public $template;
+
+  /**
+   * The pattern stylesheet.
+   *
+   * @var string
+   *   The pattern stylesheet.
+   */
+  public $stylesheet;
+
+  /**
+   * The pattern javascript.
+   *
+   * @var string
+   *   The pattern javascript.
+   */
+  public $javascript;
+
+  /**
+   * Get the pattern definition.
+   *
+   * @return string
+   *   The pattern definition.
+   */
+  public function definition() {
+    return $this->definition;
+  }
+
+  /**
+   * Get defaults definition.
+   *
+   * @param string $key
+   *   Specify a default key to get from the default definition.
+   *
+   * @return array|null
+   *   The default definition or NULL if the provided argument doesn't exists.
+   */
+  private function defaults($key = NULL) {
+    // @todo: complete this.
+    $defaults = [
```

Comment:
This defaults can be taken directly from the default properties of the UiPatternsManager:
```
\Drupal\ui_patterns\UiPatternsManager::$defaults
```